internal/cache/file/downloader/downloader_test.go

```diff
@@ -68,6 +68,19 @@ func (dt *downloaderTest) TearDown() {
 	operations.RemoveDir(cacheDir)
 }
 
+func (dt *downloaderTest) waitForCrcCheckToBeCompleted() {
+	// Last notification is sent after the entire file is downloaded and before the CRC check is done.
+	// Hence, explicitly waiting till the CRC check is done.
+	for {
+		dt.job.mu.Lock()
+		if dt.job.status.Name == Completed || dt.job.status.Name == Failed {
+			dt.job.mu.Unlock()
+			break
+		}
+		dt.job.mu.Unlock()
+	}
```

Comment:
should we keep a time gap of say a millisecond between consecutive checks ?

Reply:
Added.